### PR TITLE
[SPARK-7890] [DOCS] Document that Spark 2.11 now supports Kafka

### PR DESCRIPTION
**Pull request by srowen**
Remove caveat about Kafka / JDBC not being supported for Scala 2.11

### CODE REVIEW

**Comment by sryza** [comment]
LGTM.  Has your question on the JIRA regarding JDBC support been answered?

File: docs/building-spark.md [hunk 1]
```diff
@@ -130,10 +130,6 @@ To produce a Spark package compiled with Scala 2.11, use the `-Dscala-2.11` prop
     dev/change-version-to-2.11.sh
     mvn -Pyarn -Phadoop-2.4 -Dscala-2.11 -DskipTests clean package
 
-Scala 2.11 support in Spark does not support a few features due to dependencies
-which are themselves not Scala 2.11 ready. Specifically, Spark's external 
-Kafka library and JDBC component are not yet supported in Scala 2.11 builds.
```
Comment (pwendell) [inline docs/building-spark.md]: The JDBC server is still not supported. So actually this just needs to drop the Kafka bit.

**Comment by srowen** [comment]
@pwendell gotcha, done.

File: docs/building-spark.md [hunk 2]
```diff
@@ -130,9 +130,7 @@ To produce a Spark package compiled with Scala 2.11, use the `-Dscala-2.11` prop
...
+Scala 2.11 support in Spark does not yet support Spark's external JDBC component.
```
Comment (pwendell) [inline docs/building-spark.md]: Sorry to nitpick, but "external" doesn't make sense here (it was a modifier to Kafka support which is in our /external directory). It makes most sense to say that the JDBC/ODBC server is not supported.

Comment (srowen) [inline docs/building-spark.md]: No problem, I reworded again

**Comment by pwendell** [comment]
LGTM

**Pull request closed without merging**